classes/Product.php

```diff
@@ -642,6 +643,14 @@ class ProductCore extends ObjectModel
     public function __construct($id_product = null, $full = false, $id_lang = null, $id_shop = null, Context $context = null)
     {
         parent::__construct($id_product, $id_lang, $id_shop);
+
+        $unitPriceRatio = new Number((string) $this->unit_price_ratio);
+        $price = new Number((string) $this->price);
+
+        if ($unitPriceRatio->isGreaterThan(new Number('0'))) {
+            $this->unit_price = (float) (string) $price->dividedBy($unitPriceRatio);
```

Comment:
Since you perform the computation here shouldn't it be removed in the "full block" ? (L676)

Reply:
in that block its calculated using different price, I dont want to touch it :smile: 
(it is the price from Product::getPriceStatic).

Reply:
But maybe we should use `Product::getStaticPrice` here as well then?

Reply:
I only want to check the plain value from db. I dont see any other method to do that.
The Product::getPriceStatic is calculating the price which depends from customer, cart, address etc..

Reply:
Yeah you're right let's stick with the database value